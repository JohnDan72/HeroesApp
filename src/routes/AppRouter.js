import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';

// my components
import DashboardRouter from './DashboardRouter';
import LoginScreen from '../components/login/LoginScreen';
import NotFound from '../components/404/NotFound';

import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

const propTypes = {};
const defaultProps = {};



const AppRouter = () => {


    return (
        <Router>
            <Routes>
                <Route exact
                    path="/login"
                    element={
                        <PublicRoute>
                            <LoginScreen />
                        </PublicRoute>
                    } />
                <Route exact
                    path="/*"
                    element={
                        <PrivateRoute>
                            <DashboardRouter />
                        </PrivateRoute>
                    } />
                <Route path="*"
                    element={<NotFound />} />
            </Routes>
        </Router>

    );
}

AppRouter.propTypes = propTypes;
AppRouter.defaultProps = defaultProps;

export default AppRouter;