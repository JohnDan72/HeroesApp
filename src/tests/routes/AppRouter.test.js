import { mount, shallow } from "enzyme"
import { GeneralContext } from "../../GeneralContext";
import AppRouter from "../../routes/AppRouter"



describe('Pruebas al AppRouter', () => {
    

    test('debe renderizar el componente correctamente', () => {
        const wrapper = shallow(<AppRouter />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe renderizar login si no esta authtenticado', () => {
        const expected = 'Bienvenido a Heroes App';
        const contextValue = {
            user: {
                logged: false
            }
        }
        const wrapper = mount(
            <GeneralContext.Provider value={contextValue} >
                <AppRouter />
            </GeneralContext.Provider>
        );

        const bienvenida = wrapper.find('#id_bienvenida');
        
        expect(wrapper).toMatchSnapshot();
        expect(bienvenida.exists()).toBe(true);
        expect(bienvenida.text().trim()).toBe(expected);
    });

    test('debe renderizar cualquier private page si esta authtenticado', () => {
        const contextValue = {
            user: {
                logged: true,
                id: 23,
                name: 'Juan Daniel',
                email: 'juan1@gmail.com'
            }
        }
        const wrapper = mount(
            <GeneralContext.Provider value={contextValue} >
                <AppRouter />
            </GeneralContext.Provider>
        );

        const bienvenida = wrapper.find('#id_bienvenida');
        expect(wrapper).toMatchSnapshot();
        expect(bienvenida.exists()).toBe(false);
    });
    
})
