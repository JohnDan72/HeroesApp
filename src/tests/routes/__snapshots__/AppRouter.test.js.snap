// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al AppRouter debe renderizar el componente correctamente 1`] = `
<BrowserRouter>
  <Routes>
    <Route
      element={
        <PublicRoute>
          <LoginScreen />
        </PublicRoute>
      }
      exact={true}
      path="/login"
    />
    <Route
      element={
        <PrivateRoute>
          <DashboardRouter />
        </PrivateRoute>
      }
      exact={true}
      path="/*"
    />
    <Route
      element={<NotFound />}
      path="*"
    />
  </Routes>
</BrowserRouter>
`;
